cmd/root.go

```diff
@@ -51,54 +51,28 @@ var rootCmd = &cobra.Command{
 	Use:     "evcc",
 	Short:   "EV Charge Controller",
 	Version: server.FormattedVersion(),
-	Run:     run,
-}
-
-func bind(cmd *cobra.Command, flag string) {
-	if err := viper.BindPFlag(flag, cmd.PersistentFlags().Lookup(flag)); err != nil {
-		panic(err)
-	}
-}
-
-func configureCommand(cmd *cobra.Command) {
-	cmd.PersistentFlags().StringP(
-		"log", "l",
-		"info",
-		"Log level (fatal, error, warn, info, debug, trace)",
-	)
-	bind(cmd, "log")
-
-	cmd.PersistentFlags().StringVarP(&cfgFile,
-		"config", "c",
-		"",
-		"Config file (default \"~/evcc.yaml\" or \"/etc/evcc.yaml\")",
-	)
-	cmd.PersistentFlags().BoolP(
-		"help", "h",
-		false,
-		"Help for "+cmd.Name(),
-	)
+	Run:     runRoot,
 }
 
 func init() {
 	cobra.OnInitialize(initConfig)
-	configureCommand(rootCmd)
 
-	rootCmd.PersistentFlags().IntP("port", "p", 7070, "Listen port")
-	if err := viper.BindPFlag("network.port", rootCmd.PersistentFlags().Lookup("port")); err != nil {
-		panic(err)
-	}
+	// global options
+	rootCmd.PersistentFlags().StringVarP(&cfgFile, "config", "c", "", "Config file (default \"~/evcc.yaml\" or \"/etc/evcc.yaml\")")
+
+	rootCmd.PersistentFlags().BoolP("help", "h", false, "Help")
 
-	rootCmd.PersistentFlags().DurationP("interval", "i", 10*time.Second, "Update interval")
```

Comment:
Guten Morgen,

habe gerade den latest master bei mir deployed und diese panic bekommen

`evcc_1  | panic: non-positive interval for NewTicker
evcc_1  | goroutine 160 [running]:
evcc_1  | time.NewTicker(0xa5de22?)
evcc_1  | 	/usr/local/go/src/time/tick.go:24 +0x10f
evcc_1  | github.com/evcc-io/evcc/core.(*Site).Run(0xc0000bc900, 0xc0006fc060, 0x0)
evcc_1  | 	/build/core/site.go:540 +0x105
evcc_1  | github.com/evcc-io/evcc/cmd.runRoot.func1()
evcc_1  | 	/build/cmd/root.go:288 +0x31
evcc_1  | created by github.com/evcc-io/evcc/cmd.runRoot
evcc_1  | 	/build/cmd/root.go:287 +0x134c`

Habe den in der config das `interval: 10s` gesetzt und alles geht wieder. Ich würde sagen der default ist mit diesem cleanup verloren gegangen.

Reply:
Danke, in https://github.com/evcc-io/evcc/commit/98b8ca194d74c6444a42ef966d156394972ae8b0 gelöst